### PR TITLE
follow aws naming convention for modules #78

### LINKED ISSUE #78: follow aws naming convention for modules
Modules are built off the directory structure of the models/apis/*.json and the metadata['endpointPrefix']

Many services have names that do not match this convention (email for example is SES)

We should follow the pattern as is done in other sdks

https://github.com/aws/aws-sdk-go-v2/blob/996478f06a00c31ee7e7b0c3ac6674ce24ba0120/private/model/api/api.go#L105

### PR DESCRIPTION
**Pull request by jonnymacs**
update CodeGenerator to mimic logic in Go SDK to name modules from the api.json

https://github.com/aws/aws-sdk-go-v2/blob/996478f06a00c31ee7e7b0c3ac6674ce24ba0120/private/model/api/api.go#L105

resolves #78

### CODE REVIEW

**Comment by Yasumoto** [comment]
I just shipped https://github.com/swift-aws/aws-sdk-swift/pull/81 so I added a conflict to your `S3_API.swift` file 😬 

Any chance you could split apart the PR between the `CodeGenerator` changes and the ones created by re-running the tool? That'd be super helpful to tell apart the "meaningful" changes vs. what gets written out 🙏

**Comment by Yasumoto** [comment]
@jonnymacs sure! That'd be super helpful 🙇

**Comment by jonnymacs** [comment]
have a thorough look at this. let me know if you see anything that looks off. It compiles so that is good.

One thing to mention the Errors are now {ModuleName}ErrorType. This was to resolve a conflict with one of the apis that reserves the {ModuleName}Error classname. Now that we are consistent with naming that popped up as an issue.

Otherwise it should be win.

**Comment by jonnymacs** [comment]
@Yasumoto get a chance to review this one yet? I am making some changes it'd be cool to get this merged so I can try adding Kinesis via the module instead of the separate git repo.

File: Sources/CodeGenerator/Util.swift [hunk 1]
```diff
@@ -58,11 +58,55 @@ func loadAPIJSONList() throws -> [JSON] {
...
         var json = JSON(data: data)
-        json["serviceName"].stringValue = String($0.split(separator: "/").reversed()[2])
+        json["serviceName"].stringValue = serviceNameForApi(apiJSON: json)
         return json
     }
 }
 
+// port of https://github.com/aws/aws-sdk-go-v2/blob/996478f06a00c31ee7e7b0c3ac6674ce24ba0120/private/model/api/api.go#L105
+//
+var stripServiceNamePrefixes: [String] = [
```
Comment (Yasumoto) [inline Sources/CodeGenerator/Util.swift]: `let` to make it a constant?

File: Sources/CodeGenerator/Util.swift [hunk 2]
```diff
@@ -58,11 +58,55 @@ func loadAPIJSONList() throws -> [JSON] {
...
+  "Amazon",
+  "AWS",
+]
+
+var serviceAliases: [String:String] = [
```
Comment (Yasumoto) [inline Sources/CodeGenerator/Util.swift]: same

File: Sources/CodeGenerator/Util.swift [hunk 3]
```diff
@@ -58,11 +58,55 @@ func loadAPIJSONList() throws -> [JSON] {
...
+    var serviceName = serviceNameJSON.stringValue
+
+    serviceName.trimCharacters(in: .whitespaces)
+
+    // Strip out prefix names not reflected in service client symbol names.
+    for prefix in stripServiceNamePrefixes {
+        serviceName.deletePrefix(prefix)
+    }
+
+    // Replace all Non-letter/number values with space
```
Comment (Yasumoto) [inline Sources/CodeGenerator/Util.swift]: Isn't this deleting those characters? Might be best to delete this comment

File: Sources/CodeGenerator/main.swift [hunk 4]
```diff
@@ -15,31 +15,31 @@ var errorShapeMap: [String: String] = [:]
...
+
     group.enter()
     DispatchQueue.global().async {
         do {
-            log("Generating \(serviceName) codes ........")
-            
             let service = try AWSService(fromAPIJSON: api, docJSON: doc, endpointJSON: endpoint)
-            let basePath = "\(rootPath())/Sources/AWSSDKSwift/Services/\(api["serviceName"].stringValue)/"
+
+            log("Generating \(service.serviceName) codes ........")
```
Comment (Yasumoto) [inline Sources/CodeGenerator/main.swift]: `code`

File: Sources/CodeGenerator/Util.swift (hunk 3 above)
Comment (jonnymacs) [reply Sources/CodeGenerator/Util.swift]: yes. I lifted the comments from the go sdk which replaces non alphas with space, then removes the spaces later. Will change to `// Remove all Non-letter/number values`

**Pull request merged**